rust/worker/src/blockstore/arrow_blockfile/blockfile.rs

```diff
@@ -605,4 +652,48 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_delete() {
+        let block_provider = ArrowBlockProvider::new();
+        let mut blockfile =
+            ArrowBlockfile::new(KeyType::String, ValueType::Int32Array, block_provider);
+
+        blockfile.begin_transaction().unwrap();
+        let key1 = BlockfileKey::new("key".to_string(), Key::String("zzzz".to_string()));
+        blockfile
+            .set(
+                key1.clone(),
+                Value::Int32ArrayValue(Int32Array::from(vec![1, 2, 3])),
+            )
+            .unwrap();
+        let key2 = BlockfileKey::new("key".to_string(), Key::String("aaaa".to_string()));
+        blockfile
+            .set(
+                key2.clone(),
+                Value::Int32ArrayValue(Int32Array::from(vec![4, 5, 6])),
+            )
+            .unwrap();
+        blockfile.commit_transaction().unwrap();
+
+        blockfile.begin_transaction().unwrap();
+        blockfile.delete(key1.clone()).unwrap();
+        blockfile.commit_transaction().unwrap();
```

Comment:
Can we test rollback here as well?

Reply:
rollback is unimplemented in this PR. i have a stacked PR